test/test_autograd.py

```diff
@@ -8454,6 +8454,86 @@ def test_warning_in_backward(self, device):
         with self.assertWarnsRegex(UserWarning, "Warn from backward"):
             b.backward()
 
+class TestAllowMutationOnSaved(TestCase):
+    def tearDown(self):
+        torch.autograd.graph._cloned = dict()
+        torch.autograd.graph._use_counts = dict()
+        torch.autograd.graph._create_graph = False
```

Comment:
TODO:
- remove some outdated tearDown
- more tests
   - performing backward outside of the context the graph was created in
   - make sure the variables are cleared and no leakage
   - more tricky in-place over view testing
   - test with complex views etc.
- better way to identify when a function is a view without hardcoding it here
- error out or handle the case of multi-output views